### ISSUE TITLE
Does LCE support deployment on ARM32 Cortex-M7?

### ISSUE THREAD

**Issue by Oslomayor** [larq/compute-engine#814]
Hi all,

I am urgently seeking to deploy the TFLite models converted using Larq Compute Engine (LCE) on an ARM32 device, specifically a Cortex-M7 CPU, the STM32F7 series MCU.

I have seen some related issues, such as [#647](https://github.com/larq/compute-engine/issues/647), [#688](https://github.com/larq/compute-engine/issues/688).     
I have also noticed the "arm32" keyword in the LCE source code, [kernels_arm32.h](https://github.com/larq/compute-engine/blob/main/larq_compute_engine/core/bgemm/kernels_arm32.h).  
![image](https://github.com/user-attachments/assets/caadbbaa-cd8f-4d9b-b63b-d09969f4c962)

Does LCE support deployment on ARM32 Cortex-M7 processors? 
Is there some relevant documentation or sample code for me to reference, or are there any other solutions that would allow me to deploy BNN TFLite models on the Cortex-M7?

**Comment by Tombana** [comment]
Hi @Oslomayor.
Plumerai (the company that created Larq Compute Engine) offers an inference engine for Cortex-M CPUs, but this is a paid product and not open source. You can contact Plumerai through this form https://plumerai.com/contact_us, and the documentation of the inference engine can be found at https://docs.plumerai.com/1.16/inference_engine/ .

**Comment by Oslomayor** [comment]
Hi @Tombana 
Thank you for the information you provided. Plumerai's inference engine seems to support int8 and int16 quantized models, just as it states in the documentation: "[Plumerai's inference engine provides fast execution of INT8 quantized or INT16 quantized deep learning models in tiny memory spaces.](https://docs.plumerai.com/1.16/inference_engine/)" However, what I need is to deploy int1 models on Cortex-M CPUs.

**Comment by Tombana** [comment]
You're correct, support for 1-bit layers is not mentioned in the public documentation. However, 2D binary convolutions are supported with optimized implementations for Cortex-M4 and Cortex-M7.
